terraform/modules/infra-security-groups/main.tf

```diff
@@ -53,7 +68,184 @@ data "terraform_remote_state" "infra_networking" {
   }
 }
 
-## Resources
+resource "aws_security_group" "prometheus_alb" {
+  name        = "${var.stack_name}-prometheus-alb"
+  vpc_id      = "${data.terraform_remote_state.infra_networking.vpc_id}"
+  description = "Controls ingress and egress for prometheus ALB"
+
+  tags = "${merge(
+    local.default_tags,
+    map("Stackname", "${var.stack_name}"),
+  )}"
+}
+
+# We allow all IPs to access the ALB as Prometheus is fronted by an nginx which controls access to either approved IP
+# addresses, or users with basic auth creds
+resource "aws_security_group_rule" "ingress_from_public_http_to_prometheus_alb" {
+  security_group_id = "${aws_security_group.prometheus_alb.id}"
+  type              = "ingress"
+  from_port         = 80
+  to_port           = 80
+  protocol          = "tcp"
+  cidr_blocks       = ["0.0.0.0/0"]
+}
+
+resource "aws_security_group_rule" "ingress_from_public_https_to_prometheus_alb" {
+  security_group_id = "${aws_security_group.prometheus_alb.id}"
+  type              = "ingress"
+  from_port         = 443
+  to_port           = 443
+  protocol          = "tcp"
+  cidr_blocks       = ["0.0.0.0/0"]
+}
+
+resource "aws_security_group_rule" "egress_from_prometheus_alb_to_prometheus_ec2" {
+  security_group_id        = "${aws_security_group.prometheus_alb.id}"
+  type                     = "egress"
+  to_port                  = 80
+  from_port                = 80
+  protocol                 = "tcp"
+  source_security_group_id = "${aws_security_group.prometheus_ec2.id}"
+}
+
+resource "aws_security_group" "prometheus_ec2" {
+  name        = "${var.stack_name}-prometheus-ec2"
+  vpc_id      = "${data.terraform_remote_state.infra_networking.vpc_id}"
+  description = "Controls ingress and egress for prometheus EC2 instances"
+
+  tags = "${merge(
+    local.default_tags,
+    map("Stackname", "${var.stack_name}"),
+  )}"
+}
+
+resource "aws_security_group_rule" "ingress_from_prometheus_alb_to_prometheus_ec2" {
+  security_group_id        = "${aws_security_group.prometheus_ec2.id}"
+  type                     = "ingress"
+  to_port                  = 80
+  from_port                = 80
+  protocol                 = "tcp"
+  source_security_group_id = "${aws_security_group.prometheus_alb.id}"
+}
+
+resource "aws_security_group_rule" "ingress_from_prometheus_ec2_to_prometheus_ec2" {
+  security_group_id        = "${aws_security_group.prometheus_ec2.id}"
+  type                     = "ingress"
+  to_port                  = 9090
+  from_port                = 9090
+  protocol                 = "tcp"
+  source_security_group_id = "${aws_security_group.prometheus_ec2.id}"
+}
+
+resource "aws_security_group_rule" "ingress_from_prometheus_to_prometheus_node_exporter" {
+  security_group_id        = "${aws_security_group.prometheus_ec2.id}"
+  type                     = "ingress"
+  to_port                  = 9100
+  from_port                = 9090
```

Comment:
I don't understand why this line isn't also `9100`?

`from_port` and `to_port` define a port range, so this rule opens all ports from 9090 to 9100 inclusive, from prometheus_ec2 to itself.

Reply:
Explained understood and corrected thanks :-)